cmd/internal/flowmode/cmd_run.go

```diff
@@ -450,3 +455,32 @@ func splitPeers(s, sep string) []string {
 	}
 	return strings.Split(s, sep)
 }
+
+func setMutexBlockProfiling(l log.Logger) {
+	mutexProfileFraction := os.Getenv("MUTEX_PROFILING_PERCENT")
+	if mutexProfileFraction != "" {
+		rate, err := strconv.Atoi(mutexProfileFraction)
+		if err == nil && rate > 0 {
+			// The 100/rate is because the value is interpreted as 1/rate. So 50 would be 100/50 = 2 and become 1/2 or 50%.
+			runtime.SetMutexProfileFraction(100 / rate)
+		} else {
+			level.Error(l).Log("msg", "error setting MUTEX_PROFILING_PERCENT", "err", err, "value", mutexProfileFraction)
+			runtime.SetMutexProfileFraction(1000)
+		}
+	} else {
+		// Why 1000 because that is what istio defaults to and that seemed reasonable to start with.
+		runtime.SetMutexProfileFraction(1000)
+	}
+	blockProfileFraction := os.Getenv("BLOCK_PROFILING_PERCENT")
+	if blockProfileFraction != "" {
+		rate, err := strconv.Atoi(blockProfileFraction)
+		if err == nil && rate > 0 {
+			runtime.SetBlockProfileRate(100 / rate)
```

Comment:
This doesn't seem right; block profiling rate is _not_ the same as mutex profiling fraction; the docs for SetBlockProfileRate say that the value given to it is nanoeseconds, so here `BLOCK_PROFILING_PERCENT=50` would get block profiles every 2 nanoseconds, which is almost definitely not what the user wants.